src/calico_manifests.py

```diff
@@ -253,17 +248,29 @@ def __call__(self, obj) -> None:
 class SetEtcdSecrets(Patch):
     """A Patch class for modifying the Calico etcd Secret."""
 
+    CONFIG_MAP = {
+        "etcd-key": "client_key",
+        "etcd-cert": "client_cert",
+        "etcd-ca": "client_ca",
+    }
+
     def __call__(self, obj) -> None:
         """Modify the Calico etcd Secret by updating its data."""
         if not (obj.kind == "Secret" and obj.metadata.name == "calico-etcd-secrets"):
             return
 
+        values = {}
+        for secret_key, manifest_key in self.CONFIG_MAP.items():
+            val = self.manifests.config.get(manifest_key)
+            enc = self._encode_base64(val)
+            if enc:
+                values[secret_key] = enc
+
+        if not values:
+            log.info("Etcd secrets unavailable to patch.")
+            return
+
```

Comment:
@mateoflorido 
I ran into a `KeyError` exception during teardown of calico unit when the etcd relation was removed.